crc/services/approval_service.py

```diff
@@ -318,16 +319,20 @@ def add_approval(study_id, workflow_id, approver_uid):
         # If an existing approval request exists and no changes were made, do nothing.
         # If there is an existing approval request for a previous version of the workflow
         # then add a new request, and cancel any waiting/pending requests.
-        if latest_approval_request:
-            request_file_ids = list(file.file_data_id for file in latest_approval_request.approval_files)
+        if latest_approver_request:
+            request_file_ids = list(file.file_data_id for file in latest_approver_request.approval_files)
             current_data_file_ids.sort()
             request_file_ids.sort()
+            other_approver = latest_approval_requests.filter(ApprovalModel.approver_uid != approver_uid).first()
             if current_data_file_ids == request_file_ids:
-                return  # This approval already exists.
+                return  # This approval already exists or we're updating other approver.
             else:
-                latest_approval_request.status = ApprovalStatus.CANCELED.value
-                db.session.add(latest_approval_request)
-                version = latest_approval_request.version + 1
+                for approval_request in latest_approval_requests:
+                    if (approval_request.version == latest_approver_request.version and
+                        approval_request.status != ApprovalStatus.CANCELED.value):
```

Comment:
Maybe we should just change the pending ones to cancelled?  But I suspect this is fine for RRT.